src/mongoc/mongoc-topology-description.c

```diff
@@ -1403,9 +1443,9 @@ mongoc_topology_description_handle_ismaster (
                                               error);
 
    if (gSDAMTransitionTable[sd->type][topology->type]) {
-      TRACE("Transitioning to %d for %d", topology->type, sd->type);
+      TRACE("Transitioning to %s for %s", _mongoc_topology_description_type_string(topology), mongoc_server_description_type_string(sd));
       gSDAMTransitionTable[sd->type][topology->type] (topology, sd);
    } else {
-      TRACE("No transition entry to %d for %d", topology->type, sd->type);
+      TRACE("No transition entry to %s for %s", _mongoc_topology_description_type_string(topology), mongoc_server_description_type_string(sd));
```

Comment:
@bjori: This was the only use of the private `mongoc_server_description_type_string()` function. I created the separate ticket to track the enhancement of using string names here. Since these messages frequently appear in driver logs, it'd be very helpful if we used descriptive names instead of integers, which require counting through enum values to resolve.

> Maybe also add some assertions before dereferencing that jump table.

I assume that applies to the `if (gSDAMTransitionTable[sd->type][topology->type])` line?